services/core-api/app/api/parties/party_appt/models/mine_party_appt.py

```diff
@@ -30,54 +30,38 @@ class MinePartyAppointment(SoftDeleteMixin, AuditMixin, Base):
     processed_by = db.Column(db.String(60), server_default=FetchedValue())
     processed_on = db.Column(db.DateTime, nullable=False, server_default=FetchedValue())
 
-    #type specific foreign keys
+    # Type-specific properties
     mine_tailings_storage_facility_guid = db.Column(
         UUID(as_uuid=True),
         db.ForeignKey('mine_tailings_storage_facility.mine_tailings_storage_facility_guid'))
     permit_id = db.Column(db.Integer, db.ForeignKey('permit.permit_id'))
     permit = db.relationship('Permit', lazy='select')
+    union_rep_company = db.Column(db.String)
 
     # Relationships
     party = db.relationship('Party', lazy='joined')
-
     mine_party_appt_type = db.relationship(
         'MinePartyAppointmentType',
         backref='mine_party_appt',
         order_by='desc(MinePartyAppointmentType.display_order)',
         lazy='joined')
-
     documents = db.relationship(
         'MineDocument', lazy='joined', secondary='mine_party_appt_document_xref')
 
     def assign_related_guid(self, related_guid):
         from app.api.mines.permits.permit.models.permit import Permit
 
-        if self.mine_party_appt_type_code == "EOR":
+        if not related_guid:
+            return
+
+        if self.mine_party_appt_type_code == 'EOR':
             self.mine_tailings_storage_facility_guid = related_guid
+            self.permit = None
 
         if self.mine_party_appt_type_code in PERMIT_LINKED_CONTACT_TYPES:
             permit = Permit.find_by_permit_guid(related_guid)
-            if not permit:
-                raise AssertionError(f'Permit with guid {related_guid} not found')
-            self.permit_id = permit.permit_id
-        return
-
-    def save(self, commit=True):
```

Comment:
A lot of this validation was tidied up and moved into the individual "@validates" methods below.